src/field_theory/splitting_field.lean

```diff
@@ -94,61 +95,405 @@ by rw [nat_degree_remove_factor, hfn, n.add_sub_cancel]
 
 /-- Auxiliary construction to a splitting field of a polynomial, which removes
 `n` (arbitrarily-chosen) factors.
-
 Uses recursion on the degree. For better definitional behaviour, structures
 including `splitting_field_aux` (such as instances) should be defined using
 this recursion in each field, rather than defining the whole tuple through
 recursion.
 -/
-def splitting_field_aux (n : ℕ) : Π {K : Type u} [field K], by exactI Π (f : K[X]), Type u :=
-nat.rec_on n (λ K _ _, K) $ λ n ih K _ f, by exactI
-ih f.remove_factor
+def splitting_field_aux : Π (n : ℕ) {K : Type u} [field K], by exactI Π (f : K[X]), Type u
+| 0 K fK f := K
+| (n + 1) K fK f := by exactI splitting_field_aux n f.remove_factor
```

Comment:
What motivates the change to pattern-matching here?

Reply:
I'm not going to lie, if I remember correctly it's just because I thought it looked more neat. The code should be defeq, though. I'm happy to try and revert if that's wanted.

Reply:
Given all the other definitions still use `rec_on` I think it's probably best to be consistent.